tensorlayer/layers/core.py

```diff
@@ -60,73 +59,14 @@ def flatten_reshape(variable, name='flatten'):
     return tf.reshape(variable, shape=[-1, dim], name=name)
 
 
+@deprecated("2018-06-30", "TensorLayer relies on TensorFlow to check naming.")
 def clear_layers_name():
-    """Clear all layer names in LayersConfig
-
-    Examples
-    ---------
-    Clean the current graph and try to re-define model.
-
-    >>> for .... (different model settings):
-    >>>    with tf.Graph().as_default() as graph:   # clear all variables of TF
-    >>>       tl.layers.clear_layers_name()         # clear all layer name of TL
-    >>>       sess = tf.InteractiveSession()
-    >>>       # define and train a model here
-    >>>       sess.close()
-
-    Enable reusing layer names.
-
-    >>> net = tl.layers.InputLayer(x, name='input_layer')
-    >>> net = tl.layers.DenseLayer(net, n_units=800, name='relu1')
-    ...
-    >>> tl.layers.clear_layers_name()
-    >>> net2 = tl.layers.InputLayer(x, name='input_layer')
-    >>> net2 = tl.layers.DenseLayer(net2, n_units=800, name='relu1')
-
-    """
-    LayersConfig._layer_name_list = []
+    raise NotImplementedError('TensorLayer relies on TensorFlow to check naming. Remove this function call.')
```

Comment:
shall we not to raise exception here to provide better user experience.
We can do something like 
`logging.info('this method is DEPRECATED and  has no effect, please remove it from your code.')`
In that way we can inform users without crashing their code.